packages/vite/src/plugins/plugin.ts

```diff
@@ -38,7 +38,9 @@ export interface VitePluginOptions {
 type ViteTargets = Pick<ProjectConfiguration, 'targets' | 'metadata'>;
 
 function readTargetsCache(cachePath: string): Record<string, ViteTargets> {
-  return existsSync(cachePath) ? readJsonFile(cachePath) : {};
+  return process.env.NX_CACHE_PROJECT_GRAPH !== 'false' && existsSync(cachePath)
```

Comment:
I guess we should update the other plugins to follow this as well.

Reply:
Yeah, some plugins have this. Others don't, but they should all have it.